src/saml2/server.py

```diff
@@ -593,7 +594,11 @@ def gather_authn_response_args(self, sp_entity_id, name_id_policy, userid,
             args['best_effort'] = kwargs["best_effort"]
         except KeyError:
             args['best_effort'] = False
-
+        
+        # signing and digest algs
+        self.signing_algorithm = self.config.getattr('signing_algorithm', "idp")
```

Comment:
These variables should probably be set the same way the rest of the variables are set. So they should go in the 
`for param in [` block and get set there. Because in this method, we return the `args` not set instance properties like you currently do. Instance properties should be set in the `__init__` method where they are expected to be set. And what will happen if no `signing_algorithm` or `digest_algorithm` is set? These will be set to `None`, right? And will that work or will it fail? Maybe we want to have a default value for these two if nothing is set? If we want a default value it should be set in the `param_default` dictionary which is some lines before this one

Reply:
hehehe yes John!
In the previous revision you told me that for wouldn't match to your tastes :smile: 
No problem, that's just cosmetic stuff, I can proceed without doubts now but probably we would work a bit more to deal with the default behaviour and, if possibile, try  try to refactor many small aspects related to the definition of default alg (even outside of this PR!)

Reply:
AFAIK it works even if those params are None, uni tests told us, just as acknowledgment. I agree all the major aspect behind design choices

Reply:
> In the previous revision you told me that wouldn't match to your tastes 

Yes I hadn't done a thorough review I just noticed that line and added a comment, sorry for the confusion.
The fact that it is working is a good thing but I still would propose to have some defaults so that it's more clear what options are used. @c00kiemon5ter thoughts?

Reply:
No confution at all, I can made worse :) you're welcome, we know what's needed in time and what could be scheduled for tomorrow :sunglasses: @ivan drive this yellow submarine